ci/cloudbuild/dockerfiles/fedora-36-cmake.Dockerfile

```diff
@@ -61,10 +61,13 @@ RUN curl -sSL https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.2.tar.g
     ldconfig
 ENV PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib64/pkgconfig
 
+# We expose `absl::optional<>` in our public API. An Abseil LTS update will
+# break our `check-api` build, unless we disable the inline namespace.
 WORKDIR /var/tmp/build
 RUN curl -sSL https://github.com/abseil/abseil-cpp/archive/20220623.1.tar.gz | \
     tar -xzf - --strip-components=1 && \
     sed -i 's/^#define ABSL_OPTION_USE_\(.*\) 2/#define ABSL_OPTION_USE_\1 0/' "absl/base/options.h" && \
+    sed -i 's/^#define ABSL_OPTION_USE_INLINE_NAMESPACE 1$/#define ABSL_OPTION_USE_INLINE_NAMESPACE 0/' "absl/base/options.h" && \
```

Comment:
This gave me a lot of pause, but I think you are right.  It is fine for CI builds, we should not show this in the `demo-*` Dockerfiles.

Reply:
It gave me pause too, but the only problem I came up with was if, for some reason, we needed to use two Abseil releases at once.

I'll also assume that the distributions are different enough already that this does not affect caching for us.